Chat/AbstractSocket.h

```diff
@@ -6,7 +6,7 @@
 
 #pragma comment(lib, "Ws2_32.lib")
 
-enum
+enum SocketConnectionType
```

Comment:
as far as i know, you shouldn't explicitly set those values to enum's constants. They will have 0 and 1 values by themselves.